get/cluster_test.go

```diff
@@ -0,0 +1,138 @@
+package get
+
+import (
+	"testing"
+	"github.com/joyent/triton-kubernetes/backend/mocks"
+	"github.com/joyent/triton-kubernetes/state"
+	"github.com/spf13/viper"
+)
+
+func TestGetClusterNoClusterManager(t *testing.T) {
+	localBackend := &mocks.Backend{}
+
+	localBackend.On("States").Return([]string{}, nil)
+
+	err := GetCluster(localBackend)
+
+	expected := "No cluster managers."
+
+	if expected != err.Error() {
+		t.Error(err)
+	}
+}
+
+func TestMissingClusterManagerNonInteractiveMode(t *testing.T){
+	viper.Set("non-interactive", true)
+
+	localBackend := &mocks.Backend{}
+
+	localBackend.On("States").Return([]string{"dev-manager", "test-manager"}, nil)
+
+	err := GetCluster(localBackend)
+
+	expected := "cluster_manager must be specified"
+
+	if expected != err.Error() {
+		t.Error(err)
+	}
+
+}
+
+func TestUnidentifiedClusterManagerNonInteractiveMode(t *testing.T){
+	viper.Set("non-interactive", true)
+	viper.Set("cluster_manager", "xyz")
```

Comment:
Also, we need to clean variables from the viper object once we are done with that test as its going affect other tests.

Reply:
Okay, will add that

Reply:
I actually dont see other way in `viper` besides setting variable value to `nil` or call `viper.reset()` @nimajalali any suggestion here? Viper seems to be using `os.setEnv` internally

Reply:
viper.Reset() sets the value to nil too.